src/components/keyboard/index.jsx

```diff
@@ -1,57 +1,65 @@
-import React, { useState } from "react";
+import React, { useState, useEffect, useCallback } from "react";
 import './keyboard.css';
 import Key from '../key';
 
 export default function Keyboard({ word, layout, updateActive, checkGuess, gameOver }) {
+    // TODO: I'm pretty sure input and active can be combined into one field but i dont know how
     const [input, setInput] = useState("");
 
-    // need event in case of using the submit form button
-    function handleSubmit(event) {
-        if (event) {
-            event.preventDefault();
-        }
-        
-        if (gameOver) {
-            return;
-        }
+    function handleBackspace() {
+        const newInput = input.slice(0, -1);
+        setInput(newInput);
+        updateActive(newInput);
+    }
+
+    function handleEnter() {
         if (checkGuess(input.split(""))) {
             setInput("");
         }
     }
 
-    function onChange(event) {
-        if (gameOver) {
-            return;
+    function handleSpecialKey(key) {
+        // on screen value uses back while keyboard uses backspace
+        if(key.toLowerCase() === "back" || key.toLowerCase() === "backspace"){
+            handleBackspace();
+            return true;
         }
-        event.preventDefault();
-        const val = event.target.value.replace(/[^a-zA-Z]/g, '').toLowerCase();
-        if (val.length <= word.length) {
-            setInput(val);
-            updateActive(val);
+        if(key.toLowerCase() === "enter"){
+            handleEnter();
+            return true;
         }
+        return false;
     }
 
-    function handleClick(event) {
-        console.log(event);
-        if (event === "BACK") {
-            const newInput = input.slice(0, -1);
-            setInput(newInput);
-            updateActive(newInput);
+    function handleInput(key) {
+        if (gameOver) {
             return;
         }
+        key = key.replace(/[^a-zA-Z]/g, '').toLowerCase();
 
-        if (event === "ENTER") {
-            handleSubmit(null);
+        if (handleSpecialKey(key)) {
             return;
         }
 
-        const newInput = input + event;
+        const newInput = input + key;
         if (newInput.length <= word.length) {
             setInput(newInput);
             updateActive(newInput);
         }
     }
 
+    const handleKeydown = useCallback(event => {
+        event.preventDefault();
+        handleInput(event.key);
+    }, [input]);
```

Comment:
You need to add `handleInput` instead of `input` here. Effectively, any "controlled" thing that React generates that you wanna work with.
```
WARNING in src/components/keyboard/index.jsx
  Line 54:8:  React Hook useCallback has a missing dependency: 'handleInput'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```